src/MigrationScript.php

```diff
@@ -35,19 +35,16 @@ public static function get_id(): string {
 	 * Log the time and run the migration.
 	 *
 	 * @return MigrationRecord Data on the migration run.
-	 * @throws MigrationFailed If the migration encounters an error.
 	 */
 	public static function run(): MigrationRecord {
 		$record = MigrationRecord::start( static::class );
 
 		try {
 			static::execute( $record );
+			$record->was_success();
 		} catch ( Exception $exception ) {
-			throw new MigrationFailed(
-				'Migration ' . $record->get_migration_id() . ' failed. Message: ' . $exception->getMessage(),
-				null,
-				$exception
-			);
+			$message = 'Migration ' . $record->get_migration_id() . ' failed. Message: ' . $exception->getMessage();
```

Comment:
We probably don't want one failing script to block all the rest, so I removed throwing the exception and added the message as a log entry instead.